tests/pyret/tests/test-json.arr

```diff
@@ -0,0 +1,89 @@
+import json as J
+import string-dict as SD
+
+check "conversion":
+  p = J.read-json
+  
+  p("0") satisfies J.is-j-num
+  p('"a"') satisfies J.is-j-str
+  p("[]") satisfies J.is-j-arr
+  p("{}") satisfies J.is-j-obj
+  p("true") satisfies J.is-j-bool
+  p("false") satisfies J.is-j-bool
+  p("null") satisfies J.is-j-null
+
+  p('{"foo": 1, "baz": true}') is
+    J.j-obj([SD.string-dict: "foo", J.j-num(1), "baz", J.j-bool(true)])
+  p('[1,2,3]') is J.j-arr([list: J.j-num(1), J.j-num(2), J.j-num(3)])
+  p('[[[]]]') is J.j-arr([list: J.j-arr([list: J.j-arr([list:])])])
+  p('[5, null, {"hello": "world"}]') is
+    J.j-arr([list: J.j-num(5), J.j-null,
+      J.j-obj([SD.string-dict: "hello", J.j-str("world")])])
+end
+
+check "native":
+  n = lam(x): J.read-json(x).native() end
+
+  n("0") satisfies is-number
+  n('"a"') satisfies is-string
+  n("[]") satisfies is-empty
+  # n("{}") satisfies SD.is-string-dict # apparently there's no is-string-dict?
+  n("{}") is [SD.string-dict:]
+  n("true") satisfies is-boolean
+  n("false") satisfies is-boolean
+  n("null") satisfies is-nothing
+
+  n("5.1") is%(within-abs(0.000001)) ~5.1
```

Comment:
Is this test saying we expect this to be a roughnum, and not exactly 5.1?  I'm pretty sure this will return a Rational, per https://github.com/brownplt/pyret-lang/blob/master/src/js/base/js-numbers.js#L160.

I guess the JSON spec itself doesn't mandate a precision, but it's essentially going to be JavaScript's (double) in every reasonable application.